oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServletOAuth2AuthorizedClientExchangeFilterFunction.java

```diff
@@ -268,11 +289,41 @@ public void setAccessTokenExpiresSkew(Duration accessTokenExpiresSkew) {
 	public Mono<ClientResponse> filter(ClientRequest request, ExchangeFunction next) {
 		Optional<OAuth2AuthorizedClient> attribute = request.attribute(OAUTH2_AUTHORIZED_CLIENT_ATTR_NAME)
 				.map(OAuth2AuthorizedClient.class::cast);
-		return Mono.justOrEmpty(attribute)
-				.flatMap(authorizedClient -> authorizedClient(request, next, authorizedClient))
-				.map(authorizedClient -> bearer(request, authorizedClient))
-				.flatMap(next::exchange)
-				.switchIfEmpty(next.exchange(request));
+		if (!attribute.isPresent()) {
+			return Mono.just(request)
+					.flatMap(this::mergeRequestAttributesFromContext)
+					.filter(mergedRequest -> mergedRequest.attribute(OAUTH2_AUTHORIZED_CLIENT_ATTR_NAME).isPresent())
+					.flatMap(mergedRequest -> authorizedClient(mergedRequest, next, getOAuth2AuthorizedClient(mergedRequest.attributes())))
+					.map(authorizedClient -> bearer(request, authorizedClient))
+					.flatMap(next::exchange)
+					.switchIfEmpty(next.exchange(request));
+		} else {
+			return Mono.justOrEmpty(attribute)
+					.flatMap(authorizedClient -> authorizedClient(request, next, authorizedClient))
+					.map(authorizedClient -> bearer(request, authorizedClient))
+					.flatMap(next::exchange)
+					.switchIfEmpty(next.exchange(request));
+		}
+	}
+
+	private Mono<ClientRequest> mergeRequestAttributesFromContext(ClientRequest request) {
+		return Mono.just(ClientRequest.from(request))
+				.flatMap(builder -> Mono.subscriberContext()
+						.map(ctx -> builder.attributes(attrs -> populateRequestAttributes(attrs, ctx))))
+				.map(ClientRequest.Builder::build);
+	}
+
+	private void populateRequestAttributes(Map<String, Object> attrs, Context ctx) {
```

Comment:
Do we want `Context` to override potentially existing `attrs`?

Reply:
I original had `attrs.putIfAbsent()` but didn't see any reason for it. Do you anticipate a potential issue with overriding? The attributes in the `Context` will always be what the filter needs so I don't see any issue with this. What do you think?

Reply:
Applied `putIfAbsent()`

Reply:
My concern was if someone had a custom `Authentication` they were working with, it might be explicitly set as an attribute. Perhaps this is so that a global access token is used for client_credentials access tokens.

Reply:
I see. Ok so `putIfAbsent()` is needed...and it was updated in latest commit